pkg/util/k8sutil/self_hosted.go

```diff
@@ -109,11 +122,30 @@ func NewSelfHostedEtcdPod(m *etcdutil.Member, initialCluster, endpoints []string
 	applyPodPolicy(clusterName, pod, cs.Pod)
 	// overwrites the antiAffinity setting for self hosted cluster.
 	pod = selfHostedPodWithAntiAffinity(pod)
-
+	applyAppendHostsInitContainer(pod)
 	addOwnerRefToObject(pod.GetObjectMeta(), owner)
 	return pod
 }
 
+func applyAppendHostsInitContainer(p *v1.Pod) {
+	containerSpec := []v1.Container{
+		{
+			Name:  "append-hosts",
+			Image: "busybox",
+			Command: []string{
+				// NOTE: Init container could be re-executed on restart. We are taking the datadir as a signal of restart.
```

Comment:
what do you mean by `We are taking the datadir as a signal of restart.`?

we check the existence of datadir to determine if the container is starting or restarting?